pointmatcher/ErrorMinimizers/PointToPlane.cpp

```diff
@@ -154,7 +192,22 @@ typename PointMatcher<T>::TransformationParameters PointToPlaneErrorMinimizer<T>
 		assert(normalRef.rows() > 0);
 
 		// Compute cross product of cross = cross(reading X normalRef)
-		const Matrix cross = this->crossProduct(mPts.reading.features, normalRef);
+		Matrix cross;
+		Matrix Gamma(3,3);
+		if(!force4DOF)
+		{
+			// Compute cross product of cross = cross(reading X normalRef)
+			cross = this->crossProduct(mPts.reading.features, normalRef);
+		}
+		else
+		{
+		   	//VK: Instead for "cross" as in 3D, we need only a dot product with the Gamma factor for 4DOF
+			Gamma << 0,-1, 0,
+			         1, 0, 0,
+			         0, 0, 0;
```

Comment:
Where does this gamma factor come from? What's its physical interpretation?

If it's described in a paper, can we mention that somewhere?

Reply:
Well it is described in a bunch of wrinkled hand-written notes on my desk. I'll think how to put it somewhere.